src/main/java/org/apache/cassandra/sidecar/cdc/CDCReaderService.java

```diff
@@ -0,0 +1,157 @@
+package org.apache.cassandra.sidecar.cdc;
+
+import java.io.InvalidObjectException;
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.Host;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import org.apache.cassandra.schema.KeyspaceMetadata;
+import org.apache.cassandra.schema.Schema;
+import org.apache.cassandra.schema.TableMetadata;
+import org.apache.cassandra.sidecar.CQLSession;
+
+/**
+ * Cassandra's real-time change data capture service.
+ */
+@Singleton
+public class CDCReaderService implements Host.StateListener
+{
+    private static final Logger logger = LoggerFactory.getLogger(CDCReaderService.class);
+    private final CDCIndexWatcher cdcIndexWatcher;
+    private final CDCRawDirectoryMonitor cdcRawDirectoryMonitor;
+    private final CQLSession session;
+    private final CassandraConfig cassandraConfig;
+
+    @Inject
+    public CDCReaderService(CDCIndexWatcher cdcIndexWatcher, CDCRawDirectoryMonitor monitor, CQLSession session,
```

Comment:
CDCReaderService is injected but I do not see where the object is injected ? including CDCIndexWathcher ,CDCRawDirectoryMonitor

Reply:
To the `CassandraSidecarDaemon`?